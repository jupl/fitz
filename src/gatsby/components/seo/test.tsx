import {afterAll, beforeAll, expect, jest, test} from '@jest/globals'
import {create} from 'react-test-renderer'

void beforeAll(async () => {
  const hooks = await import('../../hooks/metadata')
  jest.spyOn(hooks, 'useGatsbyMetadata').mockReturnValue({
    author: 'Author',
    description: 'Description',
    title: 'Title',
  })
})

void afterAll(async () => {
  const hooks = await import('../../hooks/metadata')
  jest.spyOn(hooks, 'useGatsbyMetadata').mockRestore()
})

test('render', async () => {
  const React = await import('react')
  const {Helmet} = await import('react-helmet')
  const {GatsbySEO} = await import('.')
  create(<GatsbySEO />)
  expect(Helmet.peek()).toMatchSnapshot()
})
