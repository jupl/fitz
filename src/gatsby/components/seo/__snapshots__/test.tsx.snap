// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render 1`] = `
Object {
  "base": Object {
    "toComponent": [Function],
    "toString": [Function],
  },
  "bodyAttributes": Object {
    "toComponent": [Function],
    "toString": [Function],
  },
  "htmlAttributes": Object {
    "toComponent": [Function],
    "toString": [Function],
  },
  "link": Object {
    "toComponent": [Function],
    "toString": [Function],
  },
  "meta": Object {
    "toComponent": [Function],
    "toString": [Function],
  },
  "noscript": Object {
    "toComponent": [Function],
    "toString": [Function],
  },
  "script": Object {
    "toComponent": [Function],
    "toString": [Function],
  },
  "style": Object {
    "toComponent": [Function],
    "toString": [Function],
  },
  "title": Object {
    "toComponent": [Function],
    "toString": [Function],
  },
}
`;
